dvc/dependency/repo.py

```diff
@@ -1,10 +1,17 @@
+from typing import NamedTuple, Optional
+
 from voluptuous import Required
 
 from dvc.path_info import PathInfo
 
 from .base import Dependency
 
 
+class RepoPair(NamedTuple):
```

Comment:
We already were already use `RepoDependency.repo_pair` so it seemed to me that we should just have `RepoPair` as the named type. Either way, this is likely going away entirely in the future since we will move to handling external deps as a special object type (so RepoDependency will have an overridden `get_obj()` method, and then will be collected the same way as other objects)